IBM Watson voicemail transcription

konradSC opened the pull request:
First, a big shout out to bcmike over at pbxforums for doing the bulk of the work on this code. Nice work!

The following PR is for voicemail transcription using IBM Watson.
https://cloud.ibm.com/catalog/services/speech-to-text

The following values need to be configured in Default Settings: 

Category: Voicemail
Subcategory: api_key 
type: text 
Value: [Your Watson api key ]

Category: Voicemail
Subcategory: json_enabled 
type: boolean 
Value: true

Category: Voicemail
Subcategory: transcibe_language 
type: text 
Value: en-US

Category: Voicemail
Subcategory: transcribe_provider 
type: text 
Value: watson

Category: Voicemail
Subcategory: transcription_server 
type: text 
Value: https://stream.watsonplatform.net/speech-to-text/api/v1/recognize?model=en-US_NarrowbandModel

Reload Default Settings. I Flushed Cache and Reloaded XML just for good measure. 

Also make sure transcription is set to TRUE in your voicemail box.

markjcrane commented:
Would be best is bcmike signed up as a contributor and he contributed his code that way there is no question that it was donated. Then you can come in and add to it.

konradSC commented:
The code was put together by both of us. He wrote an initial version and I rewrote it to look like the above that was submitted.[ He asked me to submit the working version. ](https://www.pbxforums.com/threads/ibm-watson-integration.3397/#post-11310)

Initial version: https://www.pbxforums.com/threads/ibm-watson-integration.3397/#post-11277
Updated version: https://www.pbxforums.com/threads/ibm-watson-integration.3397/#post-11309

Only 10 lines of the 60+ lines of code are original with with bcmike. If he signed up and submitted the working code it wouldn't be right either b/c most of it is mine :-) 

Also, some of his code and my code are taken verbatim from other parts of the same lua script that blackc2004 authored back in 2016. 

I understand that you are trying to do the right thing for the project. Just let me know how you want to proceed.

konradSC commented:
Closing
